racket/src/io/time/main.rkt

```diff
@@ -0,0 +1,37 @@
+#lang racket/base
+
+(provide seconds->date)
+
+(require racket/fixnum
+         (only-in '#%kernel [date*? kernel:date*?])
+         "../common/check.rkt"
+         "../host/rktio.rkt"
+         "../string/main.rkt"
+         "../error/main.rkt")
+
+(define rktio_seconds_to_date-error-kind
+  (vector-immutable RKTIO_ERROR_KIND_RACKET
+                    RKTIO_ERROR_TIME_OUT_OF_RANGE ))
+
+(define/who seconds->date
+  (case-lambda
+   [(s) (seconds->date s #t)]
+   [(s local?)
+    (check who real? s)
+    (let* ([s (inexact->exact s)]
+           [si (floor s)]
+           [get-gmt (if local? 0 1)]
+           [nsecs (floor (* (- s si) 1000000000))]
+           [dt (rktio_seconds_to_date* rktio si nsecs get-gmt)])
```

Comment:
The general convention is to call `rktio_...` things in atomic mode, so that the operation cannot be interrupted. Any needed explicitly decallocation should happen in the atomic region, but in this case, that's built into `rktio_seconds_to_date*`. Atomic mode could be put in `rktio_seconds_to_date*`, instead.